packages/wsr-types/src/components/Sidebar.d.ts

```diff
@@ -0,0 +1,88 @@
+declare namespace __WSR {
+  namespace Sidebar {
+    export interface SidebarProps {
+      classNames?: SidebarClassNames;
+      dataHook?: string;
+      isHidden?: boolean;
+      skin?: SidebarSkin;
+      selectedKey?: string;
+    }
+
+    export class Sidebar extends React.Component<SidebarProps> {
+      static Item: typeof SidebarItem;
+      static PersistentHeader: typeof SidebarPersistentHeader;
+      static PersistentFooter: typeof SidebarPersistentFooter;
+      static BackButton: typeof SidebarBackButton;
+
+      setSelectedKey: (setSelectedKey: string) => void;
+    }
+
+    export type SidebarSkin = "dark" | "light";
+    export type SidebarClassNames = {
+      sideBar?: string;
+      content?: string;
+      slider?: string;
+      sliderOutToLeft?: string;
+      sliderOutToRight?: string;
+      sliderInFromLeft?: string;
+      sliderInFromRight?: string;
+    };
+
+    class SidebarItem extends React.PureComponent<SidebarItemProps> {}
+    interface SidebarItemProps {
+      itemKey?: string;
+      innerMenu?: React.ReactNode[];
+      disable?: boolean;
+      onClick?: (itemKey: string, e: React.MouseEvent<HTMLElement>) => void;
+    }
+
+    class SidebarPersistentHeader extends React.Component<
+      SidebarPersistentHeaderProps
+    > {}
+    interface SidebarPersistentHeaderProps {}
+
+    class SidebarPersistentFooter extends React.Component<
+      SidebarPersistentFooterProps
+    > {}
+    interface SidebarPersistentFooterProps {}
+
+    class SidebarBackButton extends React.Component<SidebarBackButtonProps> {}
+    interface SidebarBackButtonProps {
+      disable?: boolean;
+      onClick?: React.MouseEventHandler<HTMLElement>;
+    }
+
+    export class SidebarContextConsumer extends React.Component<
+      SideBarContextConsumerProps
```

Comment:
Maybe we should rename just to be aligned?

```suggestion
      SidebarContextConsumerProps
```